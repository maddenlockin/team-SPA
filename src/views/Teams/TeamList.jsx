import { useEffect, useState } from 'react'
import { Link } from 'react-router-dom';
import { deleteTeamById, getTeams } from '../../services/teams';

export default function TeamList() {
    const [teams, setTeams] = useState([]);
    const [loading, setLoading] = useState(true);
    
    const loadTeamList = async () => {
        setLoading(true);
        const response = await getTeams();
        setTeams(response);
        setLoading(false);
    }

    useEffect(() => {
        loadTeamList();
    }, []);

    const handleDelete = async ({ id, name }) => {
        const confirmation = window.confirm(`Would you like to delete ${name}?`);

        if (confirmation) {
            await deleteTeamById(id);
            await loadTeamList();
        }
    }

    if(loading)return <h1>Loading...</h1>

    return (
        <>
            <Link to='/teams/new' className='App-link'>
                Add a team
            </Link>
            <ul>
                {teams.map((team) => {
                    return <li key={team.id}>
                        <Link to={`/teams/${team.id}`} className='App-link'>
                            {team.name}
                        </Link>
                        <button
                            type='button'
                            aria-label={`Delete ${team.name}`}
                            onClick={() => handleDelete({ id: team.id, name: team.name })}
                        >
                        Delete Team
                        </button>
                    </li>
                })}
            </ul>
            
        </>
    )
}

