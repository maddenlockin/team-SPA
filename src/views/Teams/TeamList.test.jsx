import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router"
import TeamList from "./TeamList"

it('should render a list of teams', async () => {
    render(
        <MemoryRouter>
            <TeamList />
        </MemoryRouter>
    );

    const team = await screen.findByText('Kicker Bottoms');


    expect(team).toBeInTheDocument();
})