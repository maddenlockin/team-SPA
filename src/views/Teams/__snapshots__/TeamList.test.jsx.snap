// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a list of teams 1`] = `
<div>
  <ul>
    <li>
      <a
        class="App-link"
        href="/teams/1"
      >
        Bridge City Sneakers
      </a>
    </li>
    <li>
      <a
        class="App-link"
        href="/teams/2"
      >
        Kicker Bottoms
      </a>
    </li>
    <li>
      <a
        class="App-link"
        href="/teams/3"
      >
        The Ball Boys
      </a>
    </li>
  </ul>
</div>
`;
